src/main/java/org/primefaces/component/chronoline/ChronolineRenderer.java

```diff
@@ -0,0 +1,147 @@
+                .add("bottom".equals(chronoline.getAlign()), Chronoline.ALIGN_BOTTOM_CLASS)
+                .add("alternate".equals(chronoline.getAlign()), Chronoline.ALIGN_ALTERNATE_CLASS)
+                .add("horizontal".equals(chronoline.getLayout()), Chronoline.LAYOUT_HORIZONTAL_CLASS)
+                .add("vertical".equals(chronoline.getLayout()), Chronoline.LAYOUT_VERTICAL_CLASS)
+                .add(chronoline.getStyleClass())
+                .build();
+
+        writer.startElement("div", null);
+        writer.writeAttribute("id", clientId, null);
+        writer.writeAttribute("class", styleClass, null);
+        if (style != null) {
+            writer.writeAttribute("style", style, null);
+        }
+
+        encodeEvents(writer, context, chronoline);
+
+        writer.endElement("div");
+    }
+
+    protected void encodeEvents(ResponseWriter writer, FacesContext context, Chronoline chronoline) throws IOException {
+        String var = chronoline.getVar();
+        UIComponent markerFacet = chronoline.getFacet("marker");
+        UIComponent oppositeFacet = chronoline.getFacet("opposite");
+
+        Map<String, Object> requestMap = context.getExternalContext().getRequestMap();
+        Collection<?> value = (Collection<?>) chronoline.getValue();
+        if (value != null) {
+            for (Iterator<?> it = value.iterator(); it.hasNext(); ) {
+                requestMap.put(var, it.next());
+
+                writer.startElement("div", null);
+                writer.writeAttribute("class", Chronoline.EVENT_CLASS, null);
+
+                encodeOppositeContent(writer, context, oppositeFacet);
+                encodeSeparator(writer, context, markerFacet, !it.hasNext());
+
+                writer.startElement("div", null);
+                writer.writeAttribute("class", Chronoline.EVENT_CONTENT_CLASS, null);
+                renderChildren(context, chronoline);
+                writer.endElement("div");
+                writer.endElement("div");
+            }
+        }
+
+    }
+
+    protected void encodeOppositeContent(ResponseWriter writer, FacesContext context, UIComponent oppositeFacet) throws IOException {
+        writer.startElement("div", null);
+        writer.writeAttribute("class", Chronoline.EVENT_OPPOSITE_CLASS, null);
+
+        if (ComponentUtils.shouldRenderFacet(oppositeFacet)) {
+            oppositeFacet.encodeAll(context);
+        }
+
+        else {
+            writer.write("&nbsp;");
+        }
+        writer.endElement("div");
+    }
+
+    protected void encodeSeparator(ResponseWriter writer, FacesContext context, UIComponent markerFacet, boolean isLastItem) throws IOException {
+        writer.startElement("div", null);
+        writer.writeAttribute("class", Chronoline.EVENT_SEPARATOR_CLASS, null);
+
+        if (ComponentUtils.shouldRenderFacet(markerFacet)) {
+            markerFacet.encodeAll(context);
+        }
+
```

Comment:
Extra spacer line here not needed between if and else

Reply:
I agree with your opinion. Even "if else" is up to me, it should be written like this 😄 

```
if (ComponentUtils.shouldRenderFacet(oppositeFacet)) {
   oppositeFacet.encodeAll(context);
} else {
   writer.write("&nbsp;");
}
```

In fact, my ide do this automatically. However, I have to edit the code and add extra blanks every time to avoid getting a maven checkstyle error 😞